vllm/engine/arg_utils.py

```diff
@@ -707,10 +706,13 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
         )
         structured_outputs_group.add_argument(
             "--reasoning-parser",
-            # This choice is a special case because it's not static
-            choices=list(ReasoningParserManager.list_registered()),
+            # Choices need to be validated after parsing to include plugins
```

Comment:
It might have been nicer if `choices` were changed to `metavar` so that users using the CLI `--help` can still see the built in options without the passed value being strictly checked. You could add something like `<plugin>` so that users know the list in the metavar is not exhaustive.

@walterbm could this be done in a follow up?